File: Models/Abstracts/AnotacaoMusical.cs
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace Maia.Models.Abstracts
{
    public abstract class AnotacaoMusical : Entidade<int>
    {
        public long MusicaId { get; set; }
        public Musica Musica { get; set; }
        
        public int UsuarioId { get; set; }
        public Usuario Usuario { get; set; }
        
        [JsonIgnore] 
        public int QuantAcessos { get; set; } = 0;
    }
}

File: Angular/src/app/shared/models/AnotacaoMusical.ts
import {Musica} from "./Musica";
import {Usuario} from "./Usuario";

export abstract class AnotacaoMusical {
  public id?: number;
  public musicaId: number;
  public musica?: Musica;
  public usuarioId: number;
  public usuario?: Usuario;


  constructor(musicaId: number, usuarioId: number, usuario?: Usuario, musica?: Musica, id?: number) {
    this.musicaId = musicaId;
    this.musica = musica;
    this.usuarioId = usuarioId;
    this.usuario = usuario;
    this.musica = musica;
    this.id = id;
  }
}
